### ISSUE TITLE
Documentation for gopls config to enable inlay hints

### ISSUE THREAD

**Issue by omertuc** [ray-x/go.nvim#199]
Took me a while to figure this out - to get inlay hints to actually show up, you also have to configure gopls lsp with something along the lines of -

```lua 
settings = {
    gopls = {
        hints = {
            assignVariableTypes = true,
            compositeLiteralFields = true,
            compositeLiteralTypes = true,
            constantValues = true,
            functionTypeParameters = true,
            parameterNames = true,
            rangeVariableTypes = true,
        },
    },
},
```

Perhaps it should be mentioned in the README, or maybe go.nvim can do that automatically somehow instead?

**Comment by ray-x** [comment]
well.
It should added automatically 

https://github.com/ray-x/go.nvim/blob/c75824b1f050c153ebfd5be65a318b9d4463d5a9/lua/go/gopls.lua#L300-L312

**Comment by omertuc** [comment]
Interesting, I'll try to debug and see why it didn't. That piece of code is actually where I got these from but didn't realize it's supposed to happen automatically

**Comment by pertrai1** [comment]
I agree with @omertuc in that I was not able to get the hints to show up until I added those options to my lspconfig settings.

**Comment by TheLazyLemur** [comment]
Same problem here. I cannot get inlay hints to work

**Comment by danrungg** [comment]
i have the same problem
